specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json

```diff
@@ -667,25 +675,124 @@
           }
         },
         "selectorMatchOperator": {
-          "type": "string",
-          "enum": [
-            "Equals",
-            "Contains",
-            "StartsWith",
-            "EndsWith",
-            "EqualsAny"
-          ],
           "description": "When matchVariable is a collection, operate on the selector to specify which elements in the collection this exclusion applies to.",
-          "x-ms-enum": {
-            "name": "OwaspCrsExclusionEntrySelectorMatchOperator",
-            "modelAsString": true
-          }
+          "$ref": "#/definitions/MatchOperator"
         },
         "selector": {
           "type": "string",
           "description": "When matchVariable is a collection, operator used to specify which elements in the collection this exclusion applies to."
         }
       }
+    },
+    "MatchOperator": {
+      "type": "string",
+      "enum": [
+        "Equals",
+        "Contains",
+        "StartsWith",
+        "EndsWith",
+        "EqualsAny"
+      ],
+      "description": "Defines the logic for comparing two strings",
+      "x-ms-enum": {
+        "name": "OwaspCrsExclusionEntrySelectorMatchOperator",
+        "modelAsString": true
+      }
+    },
+    "BotDetection": {
+      "description": "Allows the detection of bots and specify action for each bot group",
+      "required": [
+        "good", 
+        "neutral", 
+        "bad"
+      ],
+      "properties": {
+        "good": {
+          "description": "Describes the action to be taken when detecting a good bot.",
+          "properties": {
+            "action": {
+              "description": "Which action should be triggered by the bot category",
+              "$ref": "#/definitions/Action"
+            }
+          }
+        },
+        "neutral": {
+          "description": "Describes the action to be taken when detecting a neutral bot.",
+          "properties": {
+            "action": {
+              "description": "Which action should be triggered by the bot category",
+              "$ref": "#/definitions/Action"
+            }
+          }
+        },
+        "bad": {
```

Comment:
"bad" sounds bad as a category name. I'd probably ask the AFD guys what their naming convention is since they already have the concept of bot classification